From 18d2831f2b575c2aacb299927df0a25da2edddad Mon Sep 17 00:00:00 2001
From: emwhbr <hakanbrolin@hotmail.com>
Date: Sun, 16 Jun 2019 20:52:27 +0200
Subject: [PATCH] Customize U-Boot for nolpi3

Added command gzwrite.

Added automatic boot from MMC using DTB overlay for TFT3.5.
---
 include/configs/rpi.h | 60 ++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 59 insertions(+), 1 deletion(-)

diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index 9ce41767a9..5e8e8d09bb 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -67,13 +67,15 @@
 #define CONFIG_TFTP_TSIZE
 #endif
 
+/* Various command support */
+#define CONFIG_CMD_UNZIP         /* gzwrite */
+
 /* Console configuration */
 #define CONFIG_SYS_CBSIZE		1024
 
 /* Environment */
 #define CONFIG_ENV_SIZE			SZ_16K
 #define CONFIG_SYS_LOAD_ADDR		0x1000000
-#define CONFIG_PREBOOT			"usb start"
 
 /* Shell */
 
@@ -146,6 +148,7 @@
 	"fdt_addr_r=0x02600000\0" \
 	"ramdisk_addr_r=0x02700000\0"
 
+/*
 #define BOOT_TARGET_DEVICES(func) \
 	func(MMC, mmc, 0) \
 	func(MMC, mmc, 1) \
@@ -159,6 +162,61 @@
 	ENV_DEVICE_SETTINGS \
 	ENV_MEM_LAYOUT_SETTINGS \
 	BOOTENV
+*/
 
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"bootargs_base="  \
+		"coherent_pool=1M 8250.nr_uarts=1 " \
+		"bcm2708_fb.fbwidth=656 bcm2708_fb.fbheight=416 bcm2708_fb.fbswap=1 " \
+		"vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000 " \
+		"dwc_otg.lpm_enable=0 " \
+		"rootfstype=ext4 rootwait " \
+		"console=ttyS0,115200 console=tty1 " \
+		"fbcon=map:10 fbcon=font:VGA8x8" \
+		"\0" \
+	\
+	"get_fw_bootargs="                            \
+		"fdt addr ${fdt_addr}; "                  \
+		"fdt get value bootargs /chosen bootargs" \
+		"\0"                                      \
+	\
+	"kernel_file=" \
+		"zImage"   \
+		"\0"       \
+	\
+	"dtb_file="                    \
+		"bcm2710-rpi-3-b-plus.dtb" \
+		"\0"                       \
+	\
+	"dtbo_pitft35_file="                   \
+		"/overlays/pitft35-resistive.dtbo" \
+		"\0"                               \
+	\
+	"mmc_boot="                                                             \
+		"mmc dev 0; "                                                       \
+		"echo Load DTB; "                                                   \
+		"fatload mmc 0:1 ${fdt_addr_r} ${dtb_file}; "                       \
+		"fdt addr ${fdt_addr_r}; "                                          \
+		"fdt resize; "                                                      \
+		"setexpr fdtovaddr ${fdt_addr_r} + F000; "                          \
+		"echo Load DTB overlay - TFT3.5; "                                  \
+		"fatload mmc 0:1 ${fdtovaddr} ${dtbo_pitft35_file}; "               \
+		"fdt apply ${fdtovaddr}; "                                          \
+		"fdt set /soc/spi@7e204000/pitft@0 fps <0x14>; "                    \
+		"fdt set /soc/spi@7e204000/pitft@0 spi-max-frequency <0x1312d00>; " \
+		"fdt set /soc/spi@7e204000/pitft@0 rotate <0x5a>; "                 \
+		"echo Load kernel; "                                                \
+		"fatload mmc 0:1 ${kernel_addr_r} ${kernel_file}; "                 \
+		"setenv bootargs \"${bootargs_base} root=/dev/mmcblk0p2\"; "        \
+		"echo Boot kernel; "                                                \
+		"bootz ${kernel_addr_r} - ${fdt_addr_r}"                            \
+		"\0"                                                                \
+	\
+	"bootcmd="         \
+		"run mmc_boot" \
+		"\0"           \
+	\
+	ENV_DEVICE_SETTINGS \
+	ENV_MEM_LAYOUT_SETTINGS
 
 #endif
-- 
2.17.1

